test/react-native/features/app.feature

```diff
@@ -0,0 +1,118 @@
+  Then I wait to receive a request
+  And the exception "errorClass" equals "java.lang.RuntimeException"
+  And the exception "message" equals "AppNativeHandledScenario"
+  And the event "unhandled" is false
+  And the event "app.version" equals "1.2.3"
+  And the event "app.versionCode" equals 1
+  And the event "app.releaseStage" equals "production"
+  And the event "app.inForeground" is true
+  And the event "app.duration" is not null
+  And the event "app.durationInForeground" is not null
+  And the event "app.codeBundleId" equals "1.2.3-r00110011"
+
+  # Android
+  # And the event "app.id" equals "com.reactnative"
+  And the event "app.type" equals "android"
+  # iOS
+  # And the event "app.id" equals "org.reactjs.native.example.reactnative"
+  # And the event "app.type" equals "ios"
+
+Scenario: Unhandled native error
+  When I run "AppNativeUnhandledScenario" and relaunch the app
+  And I configure Bugsnag for "AppNativeUnhandledScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "java.lang.RuntimeException"
+  And the exception "message" equals "AppNativeUnhandledScenario"
+  And the event "unhandled" is true
+  And the event "app.version" equals "1.2.3"
+  And the event "app.versionCode" equals 1
+  And the event "app.releaseStage" equals "production"
+  And the event "app.inForeground" is true
+  And the event "app.duration" is not null
+  And the event "app.durationInForeground" is not null
+  And the event "app.codeBundleId" equals "1.2.3-r00110011"
+
+  # Android
+  # And the event "app.id" equals "com.reactnative"
+  And the event "app.type" equals "android"
+  # iOS
+  # And the event "app.id" equals "org.reactjs.native.example.reactnative"
+  # And the event "app.type" equals "ios"
+
+Scenario: Setting appType in configuration
+  When I run "AppConfigAppTypeScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "AppConfigAppTypeScenario"
+  And the event "unhandled" is false
+  And the event "app.type" equals "mobileclient"
+
+Scenario: Setting releaseStage in configuration
+  When I run "AppConfigReleaseStageScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "AppConfigReleaseStageScenario"
+  And the event "unhandled" is false
+  And the event "app.releaseStage" equals "staging"
+
+Scenario: Setting releaseStage and enabledReleaseStages to enable delivery
+  When I run "AppConfigEnabledReleaseStagesScenario"
+  Then I wait to receive a request
+  And the exception "errorClass" equals "Error"
+  And the exception "message" equals "AppConfigEnabledReleaseStagesScenario"
+  And the event "unhandled" is false
+  And the event "app.releaseStage" equals "preprod"
+
+Scenario: Setting releaseStage and enabledReleaseStages to disable delivery
+  When I run "AppConfigEnabledReleaseStagesNoSendScenario"
```

Comment:
Ooops, this scenario is not defined, but it doesn't fail the tests, since it asserts that no error reports are delivered. Is there a good way to catch this?

Reply:
I think it adds weight to crafting some form of positive check in scenarios, which we continue to discuss.  We have a few ideas, including inspecting device logs for "Bugsnag loaded" and adding more logging, but it involves non-trivial updates to Maze Runner.

For this particular issue (and perhaps others), I wonder if we could use our own Bugsnag.notify in the test fixture - so that a request is sent, failing the test.  That and reviewer diligence, of course.

Reply:
ahh good idea, so in [`App.js`](https://github.com/bugsnag/bugsnag-js/pull/896/files#diff-d7f5039e8157e9d11859552ad734f896) we could do something like:

```diff
+ let Scenario = Scenarios[scenarioName]
+ if (!Scenario) {
+   Bugsnag.start(configuration)
+   Bugsnag.notify(new Error('Missing test scenario'))
+   return
+ }
- let scenario = new Scenarios[scenarioName](configuration, scenarioMetaData, jsConfig)
```

Reply:
would just have to be diligent making sure we always check the error messages for all errors received, otherwise we risk a different kind of false positive.

Reply:
Yes, absolutely, that did also occur to me.  With a bit of thought we could potentially take it further by introducing some kind of stylised error message for Maze Runner to identify and immediately fail on.